test/Common.FunctionalTests/Utilities/EventLogHelpers.cs

```diff
@@ -36,6 +43,58 @@ public static void VerifyEventLogEvent(IISDeploymentResult deploymentResult, ITe
 
             Assert.True(count > 0, $"'{expectedRegexMatchString}' didn't match any event log messaged");
             Assert.True(count < 2, $"'{expectedRegexMatchString}' matched more then one event log message");
+
+            var eventLog = new EventLog("Application");
+            var eventLogEntries = new List<EventLogEntry>();
+
+            // Perf: only get the last 20 events from the event log.
+            // Eventlog is already sorted based on time of event in ascending time.
+            // Add results in reverse order.
+            for (var i = eventLog.Entries.Count - 1; i >= eventLog.Entries.Count - 20; i--)
+            {
+                eventLogEntries.Add(eventLog.Entries[i]);
+            }
+
+            var expectedRegexEventLog = new Regex(expectedRegexMatchString);
+            var processIdString = $"Process Id: {deploymentResult.HostProcess.Id}.";
+
+            foreach (var eventLogEntry in eventLogEntries)
+            {
+                if (eventLogEntry.ReplacementStrings == null ||
+                    eventLogEntry.ReplacementStrings.Length < 3)
+                {
+                    continue;
+                }
+
+                // ReplacementStings == EventData collection in EventLog
+                // This is unaffected if event providers are not registered correctly
+                if (expectedRegexEventLog.IsMatch(eventLogEntry.ReplacementStrings[0]))
+                {
+                    if (processIdString.Contains(eventLogEntry.ReplacementStrings[1]))
+                    {
+                        if (ANCMVersionsMatch(eventLogEntry.ReplacementStrings[2], deploymentResult))
```

Comment:
This should be the first condition. No need to try and match events not from ANCM.